compiler/plugins/target/AMD-AIE/iree-amd-aie/Transforms/KernelDispatch.cpp

```diff
@@ -240,7 +257,7 @@ static LogicalResult setRootConfigForPadPackPipeline(func::FuncOp entryPointFn,
   // ------------------------------------------------------
   MLIRContext *context = entryPointFn.getContext();
   const int packLevel = 1;
-  auto packedSizes = getPackedSize(linalgOp, packLevel);
+  auto packedSizes = getPackedSize(linalgOp, packLevel, tileM1, tileN1, tileK1);
```

Comment:
Here `tilek1` is not the right thing to pass, for the 2kx2kx2k example, tilkek1 is 4 but the innerloop pack op has a source of k=16, I am trying to figure out how to statically figure that out.

Reply:
For 2kx2kx2k example, as mentioned above the inner pack size is selected fixed as (4, 4, 8). While the inner tile size for k should be set as 32 according to Erwei's experimental value. Since we have a fusion pass that fuse the pack ops into the scf.for loop, the tile size for k should be adjusted to 32/8=4, which is how 4 shows in the code.

Reply:
I am not clear how you are getting the 32 here though in the first place. As mentioned in the other comment the configuration is not changing for cases like 2kx2kx2k.
What I noticed is currently happening is. With the outer k tile of 256. The inner k tile divides that and we have a k of 256/4 = 64. So from a code-gen point of view any k packing value that divides 64  is fine, but from hardware point of view since we prefer 8, we choose that since it does divide 64. However lets say the outer size was something random like 50 needed tom support a random shape. then we will pick inner k as 2 because that's closest to 4 we can get and still divide 50, now we have a inner k of 50/2=25 and then we will choose a packing of 5, which is closest we can get to desired packing of 8 and still divide 25.

Reply:
While the inner k tile is not 64 in the previous setting. If you like we can have a chat to discuss about this.